pkg/system/phase4_configuring.go

```diff
@@ -505,10 +505,25 @@ func (r *Reconciler) generateBackingStoreTargetName() string {
 	tsMilli := strconv.FormatInt(time.Now().UnixNano()/int64(time.Millisecond), 10)
 	name := "nb." + tsMilli
 	if r.RouteMgmt.Spec.Host != "" {
-		name += "." + r.RouteMgmt.Spec.Host
+		suffix := ""
+		hostItems := strings.Split(r.RouteMgmt.Spec.Host, ".")
+		for i := len(hostItems) - 1; i >= 0; i-- {
+			hostItem := strings.Trim(hostItems[i], "-.")
+			if len(name)+1+len(hostItem)+1+len(suffix) > 63 {
```

Comment:
Suffix always includes an extra dot at the end, so this check will limit the length to one character less than the max.
This is fine, but we can also fix it.